pkg/dogstatsd/parser.go

```diff
@@ -57,20 +57,23 @@ func nextField(slice, sep []byte) ([]byte, []byte) {
 	return slice[:sepIndex], slice[sepIndex+1:]
 }
 
-// parseTags parses `rawTags` and returns a slice of tags and the value of the `host:` tag if found
-func parseTags(rawTags []byte, extractHost bool) ([]string, string) {
+// parseTags parses `rawTags` and returns a slice of tags, and, if extractHost is true,
+// whether it found the `host:` tag and its value
+func parseTags(rawTags []byte, extractHost bool) ([]string, bool, string) {
 	if len(rawTags) == 0 {
-		return nil, ""
+		return nil, false, ""
```

Comment:
Why not do `return nil, defaultHostname` here ? That would avoid returning an extra parameter.

Reply:
In `parseMetricMessage` we only assign `defaultHostname` if `foundHost` is `false` .... and that's the only time we use `foundHost`. Why not just let the function decide what hostname should be used?